### PR TITLE
batches: exclude archived, deleted changesets from completion percentage calculation

### PR DESCRIPTION
**Pull request by courier-new**
Closes https://github.com/sourcegraph/sourcegraph/issues/46830.

This was surfaced [in Slack](https://sourcegraph.slack.com/archives/CMMTWQQ49/p1674224577396439) and felt like a reasonable change.

The completion percentage is now defined based on two values: the total "actionable changesets" as the denominator, which is defined as the total minus any archived and deleted changesets; and the total "completed changesets" as the numerator, which is defined as the sum of merged and closed changesets.

There's been confusion about how the completion percentage is calculated at times before, as well as confusion over what "archived" means and how that's factored in to this percentage. For this reason, I've added an info icon + tooltip and a short description to the top of the archived changesets list, in the style of several of our admin pages where the contents of the container are explained in brief just above it. This description includes a link to the docs, too. You can see these changes below: 

image

cc @danielmarquespt

Open to wording suggestions. 🙂

## Test plan

Updated storybooks to use control knobs to test how different changeset stats affect the percentage. As in the screenshot above, you can verify archived was not included in the batch change completion percentage.

## App preview:

- [Web](https://sg-web-kr-dont-count-archived-percent.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.20 kb) | 0.01% (+1.20 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/kr/dont-count-archived-percent/compare-report.html;tab=live_object) for a full comparison between the commits d4d0e08 and 90ebc94 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/batches/detail/BatchChangeStatsCard.story.tsx [hunk 1]
```diff
@@ -39,22 +39,29 @@ export const Draft: Story = () => (
...
+                        merged: args.merged,
+                        draft: args.draft,
+                        open: args.open,
+                        total:
+                            args.closed +
+                            args.deleted +
+                            args.merged +
+                            args.draft +
+                            args.open +
+                            args.archived +
```
Comment (BolajiOlajide) [inline client/web/src/enterprise/batches/detail/BatchChangeStatsCard.story.tsx:61]:
```suggestion
                            args.merged +
                            args.draft +
                            args.open +
```

Should `archived` & `deleted` changesets still be part of the total here? I think it shouldn't.

**Review by BolajiOlajide — COMMENTED** [review]
Approving to unblock.

File: client/web/src/enterprise/batches/detail/BatchChangeDetailsTabs.tsx [hunk 2]
```diff
@@ -268,6 +268,16 @@ export const BatchChangeDetailsTabs: React.FunctionComponent<React.PropsWithChil
                     )}
                 </TabPanel>
                 <TabPanel>
+                    <Text className="my-3">
+                        Archived changesets are changesets created and published by an earlier version of the batch spec
```
Comment (eseliger) [inline client/web/src/enterprise/batches/detail/BatchChangeDetailsTabs.tsx:272]:
```suggestion
                        Archived changesets are changesets created and published by an earlier version of the batch change
```

I think this is slightly more accurate.

File: client/web/src/enterprise/batches/detail/BatchChangeStatsCard.tsx [hunk 3]
```diff
@@ -37,8 +41,11 @@ export const BatchChangeStatsCard: React.FunctionComponent<React.PropsWithChildr
     className,
 }) => {
     const { changesetsStats: stats, diffStat } = batchChange
-    const percentComplete = stats.total === 0 ? 0 : ((stats.closed + stats.merged + stats.deleted) / stats.total) * 100
-    const isCompleted = stats.closed + stats.merged + stats.deleted === stats.total
+    const percentComplete =
```
Comment (eseliger) [inline client/web/src/enterprise/batches/detail/BatchChangeStatsCard.tsx:44]: I wonder if at this point this should move in the backend, seems fairly complex logic needs to be applied in the frontend to get that value :D

**Review by eseliger — APPROVED** [review]
actually approving to unblock 😛

File: client/web/src/enterprise/batches/detail/BatchChangeStatsCard.story.tsx (hunk 1 above)
Comment (courier-new) [reply client/web/src/enterprise/batches/detail/BatchChangeStatsCard.story.tsx:61]: I believe they should be here, actually. This is supposed to mirror the data we'd get from the backend, where `total` is literally just `SELECT COUNT(*) FROM CHANGESETS`.

File: client/web/src/enterprise/batches/detail/BatchChangeStatsCard.tsx (hunk 3 above)
Comment (courier-new) [reply client/web/src/enterprise/batches/detail/BatchChangeStatsCard.tsx:44]: You're probably right. 🙂 I'm too lazy to do that right now but I'll leave a note!

**Pull request merged**